File: ios/TraceSample/TraceSample/SampleApp.swift
import SwiftUI
import TraceSDK

@main
struct SampleApp: App {
    @StateObject private var auth = AuthViewModel()

    init() {
        // Test mode — no API key needed, simulated attribution + deep links.
        // To use real credentials:
        //   let config = TraceClientConfig(
        //       apiKey:  "tr_live_your_key",
        //       hashSalt: "your_64_char_hex"
        //   )
        let config: TraceClientConfig = .test(
            simulatedDeepLink: TraceDeepLink(
                path: "/product/demo-123",
                params: ["source": "sample"],
                isDeferred: true
            )
        )
        TraceClient.initialize(config: config)
    }

    var body: some Scene {
        WindowGroup {
            TraceProvider {
                AppRootView()
                    .environmentObject(auth)
            }
        }
    }
}

File: android/src/main/kotlin/com/trace/sample/SampleApp.kt
package com.trace.sample

import android.app.Application
import com.trace.sdk.Trace
import com.trace.sdk.TraceConfig

// Test mode — no API key needed, simulated attribution + deep links.
// To use real credentials:
//   val traceConfig = TraceConfig(
//       apiKey   = "tr_live_your_key",
//       hashSalt = "your_64_char_hex",
//       region   = Region.US,
//   )
val traceConfig = TraceConfig.test()

class SampleApp : Application() {

    override fun onCreate() {
        super.onCreate()
        Trace.initialize(
            context = this,
            config = traceConfig,
        )
    }
}
